### ISSUE TITLE
[Question❓] All unknown region input

### ISSUE THREAD

**Issue by michaelowenliu** [pymatting/pymatting#55]
When I input a trimap with all unknown region, i.e. no foreground and background, it raised an error there , https://github.com/pymatting/pymatting/blob/master/pymatting/util/util.py#L491.

**Comment by 99991** [comment]
That is the intended behavior. If you have a photo of many objects, the computer can not know which object you want to cut out.

**Comment by michaelowenliu** [comment]
@99991 How about the case in the figure from [DIM paper](https://arxiv.org/pdf/1703.03872.pdf)? 
<img width="675" alt="image" src="https://user-images.githubusercontent.com/14087480/168804327-51768989-c9f7-4ec2-9bc0-b62e9b3a5184.png">

**Comment by 99991** [comment]
This library implements closed-form matting and KNN matting. As you can see from the figure, you will not get any result with an incomplete trimap.

If you only have either foreground *or* background, but not both, you can generate the other with image erosion/dilation.

Or if you only want to cut out a specific kind of object, you can cut out 100 of them by hand to make a dataset and then train a neural network like in the DIM paper.

**Comment by michaelowenliu** [comment]
@99991 The pymatting library also implements "Learning" matting, while there is a alpha matte in the paper. I plan to build a benchmark for different datasets using pymatting.

**Comment by michaelowenliu** [comment]
I also tried another implementation of CF using https://github.com/MarcoForte/closed-form-matting/blob/master/closed_form_matting/closed_form_matting.py. This implementation allows the trimap with all unknown regions, although the results are not good.

**Comment by 99991** [comment]
Learning Based Digital Matting does not use neural networks: https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.189.5195&rep=rep1&type=pdf

I am not sure why you want to benchmark different datasets. It would make more sense to benchmark different alpha matting methods.

The most comprehensive comparison of alpha matting methods is probably http://alphamatting.com/eval_25.php

There is also https://videomatting.com/ (although their HTTPS certificate is currently expired).

For a comparison with a larger dataset (Adobe Composition-1k), you can search google scholar: https://scholar.google.com/scholar?hl=en&q=composition+1k+alpha+matting

**Comment by michaelowenliu** [comment]
> Learning Based Digital Matting does not use neural networks: https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.189.5195&rep=rep1&type=pdf
> 
> I am not sure why you want to benchmark different datasets. It would make more sense to benchmark different alpha matting methods.
> 
> The most comprehensive comparison of alpha matting methods is probably http://alphamatting.com/eval_25.php
> 
> There is also https://videomatting.com/ (although their HTTPS certificate is currently expired).
> 
> For a comparison with a larger dataset (Adobe Composition-1k), you can search google scholar: https://scholar.google.com/scholar?hl=en&q=composition+1k+alpha+matting

@99991 Thanks a lot for the clear explanation. Yes, I know 'Learning Based Digital Matting does not use neural networks'. What I mean is that the original method can handle the trimap with all unknown regions, but the method implemented in pymatting cannot. How can I use this behavior that allows unknown trimap? Thanks.

**Comment by 99991** [comment]
> I also tried another implementation of CF using https://github.com/MarcoForte/closed-form-matting/blob/master/closed_form_matting/closed_form_matting.py. This implementation allows the trimap with all unknown regions, although the results are not good.

It was not meant to work with such trimaps, so it is not surprising that the results are not good.

> What I mean is that the original method can handle the trimap with all unknown regions

What makes you think so?

**Comment by michaelowenliu** [comment]
> What makes you think so?
@99991 The result in the paper.

<img width="1281" alt="image" src="https://user-images.githubusercontent.com/14087480/168819059-2b954b19-7bd4-45e0-9d28-b45517617d1c.png">

<img width="452" alt="image" src="https://user-images.githubusercontent.com/14087480/168819143-0efb3ef2-6bde-474a-8fe4-127334991a00.png">

**Comment by 99991** [comment]
That figure shows that all methods except DIM do not work for this trimap. The authors say so in the last paragraph of section 5.2

**Comment by michaelowenliu** [comment]
@99991 ok, I see. Thanks a lot.
